lib/sda/overlay.go

```diff
@@ -231,26 +238,40 @@ func (o *Overlay) Close() {
 	}
 }
 
-// CreateProtocol returns a fresh Protocol Instance with an attached
-// TreeNodeInstance
-func (o *Overlay) CreateProtocol(t *Tree, name string) (ProtocolInstance, error) {
-	tni := o.NewTreeNodeInstanceFromProtoName(t, name)
-	pi, err := ProtocolInstantiate(ProtocolNameToID(name), tni)
+// CreateProtocolSDA returns a fresh Protocol Instance with an attached
+// TreeNodeInstance. This protocol won't be handled by the service, but
+// only by the SDA.
+func (o *Overlay) CreateProtocolSDA(t *Tree, name string) (ProtocolInstance, error) {
+	return o.CreateProtocolService(ServiceID(uuid.Nil), t, name)
+}
+
+// CreateProtocolService adds the service-id to the token so the protocol will
+// be picked up by the correct service and handled by its NewProtocol method.
+func (o *Overlay) CreateProtocolService(sid ServiceID, t *Tree, name string) (ProtocolInstance, error) {
+	tni := o.NewTreeNodeInstanceFromService(t, t.Root, ProtocolNameToID(name), sid)
+	pi, err := ProtocolInstantiate(tni.token.ProtoID, tni)
 	if err != nil {
 		return nil, err
 	}
-	o.RegisterProtocolInstance(pi)
+	if err = o.RegisterProtocolInstance(pi); err != nil {
+		return nil, err
+	}
 	go pi.Dispatch()
 	return pi, err
 }
 
 // StartProtocol will create and start a P.I.
 func (o *Overlay) StartProtocol(t *Tree, name string) (ProtocolInstance, error) {
-	pi, err := o.CreateProtocol(t, name)
+	pi, err := o.CreateProtocolSDA(t, name)
 	if err != nil {
 		return nil, err
 	}
-	go pi.Start()
+	go func() {
+		err := pi.Start()
+		if err != nil {
+			dbg.Error("Error while starting:", err)
```

Comment:
Having that only printed and not returned in a function named `StartProtocol` seems to me that we should not expect an error value return in `Start`, because we don't / can't use it.